File: capgemini/batch-2/03-02-24/SlidingWindow.py
arr = [2,-3,23,-5,-6,65,-32,-1,2,-3,-23,1,5,-6]
k = 3
i = 0
j = k-1
ans = float("-inf")
ps = 0
while j<len(arr):
    if i == 0:
        currSum = sum(arr[i:j+1])
        ans = max(ans,currSum)
        ps = currSum
    else:
        currSum = ps-arr[i-1]+arr[j]
        ans = max(ans,currSum)
        ps = currSum
    i+=1
    j+=1
print(ans)

File: PDCET/22-02-2024/SlidingWindow.java
public class SlidingWindow {
    public static void main(String[] args) {
        int[] arr = {3,-1,2,7,9,-5,4,12};
        int k = 3;
        int ans = Integer.MIN_VALUE;
        int i=0, j = k-1;
        int prevSum = 0;
        while (j<arr.length) {
            int currSum = 0;
            if (i==0) {
                for(int l=i;l<=j;l++) currSum+=arr[l];
                prevSum = currSum;
            }else{
                currSum = prevSum-arr[i-1]+arr[j];
                prevSum = currSum;
            }
            i+=1;
            j+=1;
            ans = Math.max(ans,currSum);
        }
        System.out.println(ans);
    }
}
